File: AEDS_II/TPs/TP_1/TP01Q23 - Palindromo em Python/Palindromo.py
def is_palindromo(palavra):
    tamanho = len(palavra)
    for i in range(tamanho // 2):  # percorre até a metade
        if palavra[i] != palavra[tamanho - i - 1]:
            return False
    return True


def main():
    while True:
        palavra = input().strip()  # lê a palavra
        if palavra.upper() == "FIM":  # condição de parada
            break
        resultado = is_palindromo(palavra)
        print("SIM" if resultado else "NAO")


if __name__ == "__main__":
    main()

File: AEDS_II/TPs/TP_1/TP01Q25 - Palindromo em C#/Palindromo.cs
using System;

class Palindromo
{
    static bool EhPalindromo(string palavra)
    {
        int tamanho = palavra.Length;
        for (int i = 0; i < tamanho / 2; i++)
        {
            if (palavra[i] != palavra[tamanho - i - 1])
            {
                return false;
            }
        }
        return true;
    }

    static void Main()
    {
        while (true)
        {
            string palavra = Console.ReadLine().Trim();
            if (palavra.ToUpper() == "FIM")
                break;

            bool resultado = EhPalindromo(palavra);
            Console.WriteLine(resultado ? "SIM" : "NAO");
        }
    }
}
